gateway-server/src/main/java/org/apache/knox/gateway/websockets/GatewayWebsocketHandler.java

```diff
@@ -173,12 +174,41 @@ private synchronized String getMatchedBackendURL(final String path) {
 
     /* URL used to connect to websocket backend */
     String backendURL = urlFromServiceDefinition(serviceRegistryService, entry, path);
+    LOG.logMessage("Url obtained from services definition: " + backendURL);
 
     StringBuilder backend = new StringBuilder();
     try {
-
-      /* if we do not find websocket URL we default to HTTP */
-      if (!StringUtils.containsAny(backendURL, WEBSOCKET_PROTOCOL_STRING, SECURE_WEBSOCKET_PROTOCOL_STRING)) {
+      if (StringUtils.containsAny(backendURL, WEBSOCKET_PROTOCOL_STRING, SECURE_WEBSOCKET_PROTOCOL_STRING)) {
```

Comment:
Seems like there is a lot of copy/paste here. Based on the description I expected a smaller change basically checking if there is a starting/trailing slash.

Reply:
Yeah, as mentioned in my checkin commit, I added a small enhancement to proxy websocket request even if service protocol has 'http/https' in Role definition. Mentioned the same as comment on Jira as well.

Reply:
Yea sure, but inside the two if clauses there are a bunch of duplicate lines. Especially around handling `/ws` and `/websocket` which I think should be in one place. Handling the prefix/suffix shouldn't need to happen in two places.

Reply:
Like @risdenk mentioned there is a lot of code duplication and rearranging. Let's try not to rearrange code unless it is necessary.

Reply:
Thanks for useful comments. Removed code duplication as pointed out bu @risdenk . Please check.